staging/src/k8s.io/apiserver/pkg/util/proxy/proxy.go

```diff
@@ -29,51 +29,103 @@ import (
 	"k8s.io/client-go/pkg/api/v1"
 )
 
-// ResourceLocation returns a URL to which one can send traffic for the specified service.
-func ResolveEndpoint(services listersv1.ServiceLister, endpoints listersv1.EndpointsLister, namespace, id string) (*url.URL, error) {
+func parseID(id string) (scheme, name, port string, err error) {
+	if len(id) == 0 {
+		return "", "", "", fmt.Errorf("invalid empty service name %q", id)
+	}
+
 	// Allow ID as "svcname", "svcname:port", or "scheme:svcname:port".
 	svcScheme, svcName, portStr, valid := utilnet.SplitSchemeNamePort(id)
 	if !valid {
-		return nil, errors.NewBadRequest(fmt.Sprintf("invalid service request %q", id))
+		return "", "", "", errors.NewBadRequest(fmt.Sprintf("invalid service request %q", id))
+	}
+
+	if len(svcScheme) == 0 {
+		svcScheme = "https"
+	}
+	if len(portStr) == 0 {
+		switch svcScheme {
+		case "http":
```

Comment:
My argument for http was that it's useful during development without certs. But if we can ignore self-signed certs anyway, there is not much need for http.